app/models/reminder.rb

```diff
@@ -51,57 +51,54 @@ def self.create_for(kid, time)
   # - not all journal entries or reminders are there
   # - you pass a parater time that is a few days after the kids meeting das
   def self.conditionally_create_reminders(time = Time.now)
-    reminders_created = 0
-
     logger.info("Beginning reminder run, reference Date #{time}")
     logger.flush
 
-    begin
-
-      Kid.all.find_each do |kid|
-        logger.info("[#{kid.id}] #{kid.display_name}: checking journal entries")
-        logger.flush
-        unless kid.journal_entry_due?(time)
-          logger.info("[#{kid.id}] #{kid.display_name}: no entry due")
-          logger.flush
-          next
-        end
-        if kid.journal_entry_for_week(time)
-          logger.info("[#{kid.id}] #{kid.display_name}: journal entry present")
-          logger.flush
-          next
-        end
-        if kid.reminder_entry_for_week(time)
-          logger.info("[#{kid.id}] #{kid.display_name}: reminder entry present")
-          logger.flush
-          next
-        end
-        if kid.mentor.nil? && kid.secondary_mentor.nil?
-          logger.info("[#{kid.id}] #{kid.display_name}: no mentors set")
-          logger.flush
-          next
-        end
-        reminder = Reminder.create_for(kid, time)
-        logger.info("[#{kid.id}] #{kid.display_name}: created reminder [#{reminder.id}]")
-        logger.flush
-        reminders_created += 1
-      end
+    reminders_created_count =
+      Kid.all.find_each.select do |kid|
+        self.conditionally_create_reminder_for_kid(time, kid)
+      end.count
```

Comment:
You can pass a block to `#count` (instead of doing `#select` followed by `#count`) and it will only count the items for which the block evaluates to true.